lib/src/js/polymer_base_dart.html

```diff
@@ -75,27 +75,38 @@
       // Secret hook into property changes. Pretty hacky but its more efficient
       // than using a JsProxy object for the element.
       _propertyChanged: function(path, newValue, oldValue) {
-        var parts = path.split('.');
-        var thisArg = parts.length == 1 ?
-            this : this.get(parts.slice(0, parts.length - 1));
-        thisArg = thisArg.__dartClass__ || thisArg;
-        Polymer.Dart.propertyChanged(
-            thisArg, parts[parts.length - 1], newValue);
-      },
+        var parts = this._getPathParts(path);
+        var array;
 
-      notifyPath: function(path, value, fromAbove) {
-        var notified =
-            Polymer.Base.notifyPath.call(this, path, value, fromAbove);
-        if (fromAbove) return notified;
+        // Get the model which is being updated.
+        var thisArg = this;
+        for (var i = 0; i < parts.length - 1; i++) {
+          var part = parts[i];
 
-        // If you call notifyPath from dart we need to do manual modifications
-        // on any JsArray or JsObject instances.
-        var parts = this._getPathParts(path);
-        if (parts.length > 1) {
-          var model = this.get(parts.splice(0, parts.length - 1));
-          if (model[parts[0]] != value) model[parts[0]] = value;
+          if (array && (parseInt(part) == part)) {
```

Comment:
Actually I found that when replacing an element in an array it will behave
just like an addition and a removal thus not notifying a path ending with
key even in such a use case. Didn't find a way to create such a use case so
far.

2015-09-30 23:46 GMT+02:00 Jacob MacDonald <notifications@github.com>:

> In lib/src/js/polymer_base_dart.html
> <https://github.com/dart-lang/polymer-dart/pull/602#discussion_r40856736>:
>
> >
> > -        // If you call notifyPath from dart we need to do manual modifications
> > -        // on any JsArray or JsObject instances.
> > -        var parts = this._getPathParts(path);
> > -        if (parts.length > 1) {
> > -          var model = this.get(parts.splice(0, parts.length - 1));
> > -          if (model[parts[0]] != value) model[parts[0]] = value;
> > +          if (array && (parseInt(part) == part)) {
>
> This is how they do it in set, and other places, I think its worth
> keeping just for consistency. Really there should be a helper method in
> polymer js or something.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dart-lang/polymer-dart/pull/602/files#r40856736>.
>